tests/add.sh

```diff
@@ -1,4 +1,7 @@
-source common.sh
+export NIX_TEST_ROOT="$(cd -P -- "$(dirname -- "$0")" && pwd -P)"
```

Comment:
This prefix, common to all the tests, would allow us to run each individual test from a distinct directory

common.sh would no longer be generated. We would then expect the scripts to work on an installed copy of nix, without having these scripts rely on any "baking" (substitutions from the build system).